pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -96,11 +98,45 @@ func initContainerName(lang language) string {
 	return fmt.Sprintf("datadog-lib-%s-init", lang)
 }
 
-func injectAutoInstrumentation(pod *corev1.Pod, _ string, _ dynamic.Interface) error {
+// getDeploymentReference returns the OwnerReference if it is of Kind Deployment
+func getDeploymentReference(pod *corev1.Pod) *v1.OwnerReference {
+	ownerReferences := pod.ObjectMeta.OwnerReferences
+
+	for _, ownerRef := range ownerReferences {
+		if ownerRef.Kind == "Deployment" {
+			return &ownerRef
+		}
+	}
+	return nil
+}
+
+func injectAutoInstrumentation(pod *corev1.Pod, _ string, dc dynamic.Interface) error {
 	if pod == nil {
 		return errors.New("cannot inject lib into nil pod")
 	}
 
+	if config.Datadog.GetBool("admission_controller.auto_instrumentation.apm_instrumentation_enabled") {
+		if deployment := getDeploymentReference(pod); deployment == nil {
+			log.Debugf("Skipping pod %q as it's not from a deployment", podString(pod))
+			return nil
+		}
+
+		// The way to opt-out of the automatic instrumentation is to set the enabled label to false.
+		if val, found := pod.GetLabels()[admCommon.EnabledLabelKey]; found {
```

Comment:
I'm pretty sure this works, but it's a little weird. You only need to worry about "found" if you need to distinguish between a zero value that's stored in a map and a zero value that's returned because the key isn't present. The zero value for a string is "", so there's no possible confusion. 

The following code would be simpler and should be equivalent:

```go
if pod.GetLabels()[admCommon.EnabledLabelKey] == "false" {
    log.Debugf("Skipping auto instrumentation of pod %q due to label", podString(pod))
    return nil
}
```